src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -313,12 +473,14 @@ namespace OpenMS
       }
 
       double avgFWHM { 0 };
-      for (Size j = 0; j < picked_spectra[i].getFloatDataArrays()[0].size(); ++j)
+      if (picked_spectra[i].getFloatDataArrays().size())
```

Comment:
Can you use !empty here?

Reply:
And in the following instances where size != 0 is used to implicitly ask for !empty.